File: website/shell/renderer.js
/**
 * Card Renderer for Seabay Shell Web
 *
 * Renders Seabay cards (task_approval, match_result, agent profiles, inbox)
 * as interactive HTML elements.
 */

class SeabayRenderer {
  /**
   * Render a task approval card.
   * @param {object} task
   * @param {object} callbacks - { onAccept(taskId), onDecline(taskId) }
   * @returns {HTMLElement}
   */
  static taskApprovalCard(task, callbacks = {}) {
    const card = document.createElement('div');
    card.className = 'card';
    card.innerHTML = `
      <div class="card-header">
        <span>Task Approval Required</span>
        <span class="badge risk-${(task.risk_level || 'r0').toLowerCase()}">${task.risk_level || 'R0'}</span>
      </div>
      <div class="card-body">
        <div class="card-field">
          <span class="label">Task ID</span>
          <span class="value">${SeabayRenderer.esc(task.id || '?')}</span>
        </div>
        <div class="card-field">
          <span class="label">Type</span>
          <span class="value">${SeabayRenderer.esc(task.task_type || '?')}</span>
        </div>
        <div class="card-field">
          <span class="label">From</span>
          <span class="value">${SeabayRenderer.esc(task.from_agent_id || '?')}</span>
        </div>
        ${task.description ? `
        <div class="card-field">
          <span class="label">Description</span>
          <span class="value">${SeabayRenderer.esc(task.description)}</span>
        </div>` : ''}
        <div class="card-field">
          <span class="label">Status</span>
          <span class="value"><span class="badge status-pending">${SeabayRenderer.esc(task.status || 'pending')}</span></span>
        </div>
      </div>
      <div class="card-actions">
        <button class="btn success accept-btn">Accept</button>
        <button class="btn danger decline-btn">Decline</button>
      </div>
    `;

    const acceptBtn = card.querySelector('.accept-btn');
    const declineBtn = card.querySelector('.decline-btn');

    acceptBtn.addEventListener('click', () => {
      acceptBtn.disabled = true;
      declineBtn.disabled = true;
      acceptBtn.textContent = 'Accepting...';
      if (callbacks.onAccept) callbacks.onAccept(task.id);
    });

    declineBtn.addEventListener('click', () => {
      acceptBtn.disabled = true;
      declineBtn.disabled = true;
      declineBtn.textContent = 'Declining...';
      if (callbacks.onDecline) callbacks.onDecline(task.id);
    });

    return card;
  }

  /**
   * Render match results with selectable candidates.
   * @param {Array} agents
   * @param {string} summary
   * @param {function} onSelect - callback(agentId)
   * @returns {HTMLElement}
   */
  static matchResultCard(agents, summary = '', onSelect = null) {
    const container = document.createElement('div');

    if (summary) {
      const summaryEl = document.createElement('p');
      summaryEl.style.marginBottom = '8px';
      summaryEl.textContent = summary;
      container.appendChild(summaryEl);
    }

    if (!agents || agents.length === 0) {
      const empty = document.createElement('p');
      empty.style.color = 'var(--text-secondary)';
      empty.textContent = 'No matches found.';
      container.appendChild(empty);
      return container;
    }

    agents.forEach((agent, idx) => {
      const card = document.createElement('div');
      card.className = 'card';
      card.style.marginBottom = '8px';

      const skills = (agent.skills || agent.profile?.skills || []);
      const skillsHtml = skills.length
        ? `<div class="skills-row">${skills.map(s => `<span class="skill-tag">${SeabayRenderer.esc(s)}</span>`).join('')}</div>`
        : '';

      const reasons = agent.why_matched || agent.reasons || [];
      const reasonsHtml = reasons.length
        ? `<ul class="reason-list">${reasons.map(r => `<li>${SeabayRenderer.esc(r)}</li>`).join('')}</ul>`
        : '';

      const agentRef = agent.agent_ref || agent.agent_id || agent.id || '?';
      const displayName = agent.display_name || 'Unknown';
      const verification = agent.verification || agent.verification_level || 'none';

      card.innerHTML = `
        <div class="card-header">
          <span>#${idx + 1} ${SeabayRenderer.esc(displayName)}</span>
          <span class="badge status-${agent.status === 'online' ? 'online' : 'offline'}">${SeabayRenderer.esc(verification)}</span>
        </div>
        <div class="card-body">
          <div class="card-field">
            <span class="label">Agent</span>
            <span class="value">${SeabayRenderer.esc(agentRef)}</span>
          </div>
          ${agent.location ? `<div class="card-field"><span class="label">Location</span><span class="value">${SeabayRenderer.esc(agent.location)}</span></div>` : ''}
          ${agent.last_active ? `<div class="card-field"><span class="label">Last Active</span><span class="value">${SeabayRenderer.esc(agent.last_active)}</span></div>` : ''}
          ${agent.success_rate_30d != null ? `<div class="card-field"><span class="label">Success Rate</span><span class="value">${Math.round(agent.success_rate_30d * 100)}%</span></div>` : ''}
          ${agent.match_score != null ? `<div class="card-field"><span class="label">Score</span><span class="value">${agent.match_score.toFixed(1)}</span></div>` : ''}
          ${skillsHtml}
          ${reasonsHtml}
        </div>
        ${onSelect ? '<div class="card-actions"><button class="btn primary select-btn">Select This Agent</button></div>' : ''}
      `;

      if (onSelect) {
        card.querySelector('.select-btn').addEventListener('click', () => {
          onSelect(agentRef);
        });
      }

      container.appendChild(card);
    });

    return container;
  }

  /**
   * Render a task detail card.
   * @param {object} task
   * @returns {HTMLElement}
   */
  static taskDetailCard(task) {
    const card = document.createElement('div');
    card.className = 'card';
    card.innerHTML = `
      <div class="card-header">
        <span>Task Details</span>
        <span class="badge risk-${(task.risk_level || 'r0').toLowerCase()}">${task.risk_level || 'R0'}</span>
      </div>
      <div class="card-body">
        <div class="card-field"><span class="label">Task ID</span><span class="value">${SeabayRenderer.esc(task.id || '?')}</span></div>
        <div class="card-field"><span class="label">Type</span><span class="value">${SeabayRenderer.esc(task.task_type || '?')}</span></div>
        <div class="card-field"><span class="label">Status</span><span class="value"><span class="badge status-${SeabayRenderer.statusClass(task.status)}">${SeabayRenderer.esc(task.status || '?')}</span></span></div>
        <div class="card-field"><span class="label">From</span><span class="value">${SeabayRenderer.esc(task.from_agent_id || '?')}</span></div>
        <div class="card-field"><span class="label">To</span><span class="value">${SeabayRenderer.esc(task.to_agent_id || '?')}</span></div>
        ${task.description ? `<div class="card-field"><span class="label">Description</span><span class="value">${SeabayRenderer.esc(task.description)}</span></div>` : ''}
        ${task.created_at ? `<div class="card-field"><span class="label">Created</span><span class="value">${SeabayRenderer.esc(task.created_at)}</span></div>` : ''}
      </div>
    `;
    return card;
  }

  /**
   * Render an agent profile card.
   * @param {object} agent
   * @returns {HTMLElement}
   */
  static agentProfileCard(agent) {
    const profile = agent.profile || {};
    const skills = profile.skills || [];
    const languages = profile.languages || [];

    const card = document.createElement('div');
    card.className = 'card';

    const skillsHtml = skills.length
      ? `<div class="skills-row">${skills.map(s => `<span class="skill-tag">${SeabayRenderer.esc(s)}</span>`).join('')}</div>`
      : '';

    card.innerHTML = `
      <div class="card-header">
        <span>${SeabayRenderer.esc(agent.display_name || 'Unknown')}</span>
        <span class="badge status-${agent.status === 'online' ? 'online' : 'offline'}">${SeabayRenderer.esc(agent.status || 'offline')}</span>
      </div>
      <div class="card-body">
        <div class="card-field"><span class="label">Slug</span><span class="value">@${SeabayRenderer.esc(agent.slug || '?')}</span></div>
        <div class="card-field"><span class="label">Type</span><span class="value">${SeabayRenderer.esc(agent.agent_type || '?')}</span></div>
        <div class="card-field"><span class="label">Verification</span><span class="value">${SeabayRenderer.esc(agent.verification_level || 'none')}</span></div>
        ${profile.bio ? `<div class="card-field"><span class="label">Bio</span><span class="value">${SeabayRenderer.esc(profile.bio)}</span></div>` : ''}
        ${profile.location_city ? `<div class="card-field"><span class="label">Location</span><span class="value">${SeabayRenderer.esc(profile.location_city)}${profile.location_country ? ', ' + SeabayRenderer.esc(profile.location_country) : ''}</span></div>` : ''}
        ${languages.length ? `<div class="card-field"><span class="label">Languages</span><span class="value">${SeabayRenderer.esc(languages.join(', '))}</span></div>` : ''}
        ${skillsHtml}
      </div>
    `;
    return card;
  }

  /**
   * Render inbox task list.
   * @param {Array} tasks
   * @param {boolean} hasMore
   * @param {object} callbacks - { onAccept(id), onDecline(id) }
   * @returns {HTMLElement}
   */
  static inboxCard(tasks, hasMore = false, callbacks = {}) {
    const container = document.createElement('div');

    if (!tasks || tasks.length === 0) {
      const empty = document.createElement('p');
      empty.style.color = 'var(--text-secondary)';
      empty.textContent = 'Inbox is empty.';
      container.appendChild(empty);
      return container;
    }

    tasks.forEach(task => {
      const card = document.createElement('div');
      card.className = 'card';
      card.style.marginBottom = '8px';

      const needsAction = task.status === 'pending_accept';

      card.innerHTML = `
        <div class="card-body" style="padding: 10px 14px;">
          <div style="display: flex; justify-content: space-between; align-items: center;">
            <div>
              <strong>${SeabayRenderer.esc(task.task_type || '?')}</strong>
              <span class="badge risk-${(task.risk_level || 'r0').toLowerCase()}" style="margin-left: 6px;">${task.risk_level || 'R0'}</span>
            </div>
            <span class="badge status-${SeabayRenderer.statusClass(task.status)}">${SeabayRenderer.esc(task.status || '?')}</span>
          </div>
          ${task.description ? `<p style="font-size: 12px; color: var(--text-secondary); margin-top: 4px;">${SeabayRenderer.esc(task.description.substring(0, 100))}</p>` : ''}
        </div>
        ${needsAction ? `
        <div class="card-actions">
          <button class="btn success accept-btn" style="font-size: 11px; padding: 4px 12px;">Accept</button>
          <button class="btn danger decline-btn" style="font-size: 11px; padding: 4px 12px;">Decline</button>
        </div>` : ''}
      `;

      if (needsAction) {
        card.querySelector('.accept-btn')?.addEventListener('click', () => {
          if (callbacks.onAccept) callbacks.onAccept(task.id);
        });
        card.querySelector('.decline-btn')?.addEventListener('click', () => {
          if (callbacks.onDecline) callbacks.onDecline(task.id);
        });
      }

      container.appendChild(card);
    });

    if (hasMore) {
      const more = document.createElement('p');
      more.style.cssText = 'font-size: 12px; color: var(--text-secondary); text-align: center; margin-top: 8px;';
      more.textContent = '... more tasks available';
      container.appendChild(more);
    }

    return container;
  }

  // ── Utility ──

  static esc(str) {
    const div = document.createElement('div');
    div.textContent = str;
    return div.innerHTML;
  }

  static statusClass(status) {
    if (['completed'].includes(status)) return 'completed';
    if (['pending_accept', 'pending_delivery', 'in_progress', 'accepted', 'delivered'].includes(status)) return 'pending';
    if (['failed', 'declined'].includes(status)) return 'failed';
    return 'offline';
  }
}

File: shell-cli/seabay_shell/renderer.py
"""Terminal card rendering for Seabay Shell — renders cards and structured data with ANSI colors."""

from __future__ import annotations

from typing import Optional

try:
    from rich.console import Console
    from rich.panel import Panel
    from rich.table import Table
    from rich.markdown import Markdown

    HAS_RICH = True
except ImportError:
    HAS_RICH = False


# ANSI color codes as fallback
class _Colors:
    RESET = "\033[0m"
    BOLD = "\033[1m"
    DIM = "\033[2m"
    RED = "\033[31m"
    GREEN = "\033[32m"
    YELLOW = "\033[33m"
    BLUE = "\033[34m"
    MAGENTA = "\033[35m"
    CYAN = "\033[36m"
    WHITE = "\033[37m"
    BG_RED = "\033[41m"
    BG_GREEN = "\033[42m"
    BG_YELLOW = "\033[43m"
    BG_BLUE = "\033[44m"


class TerminalRenderer:
    """Renders Seabay cards, task info, and match results in the terminal."""

    def __init__(self, use_rich: bool = True):
        self.use_rich = use_rich and HAS_RICH
        if self.use_rich:
            self.console = Console()
        else:
            self.console = None

    def render_text(self, text: str) -> None:
        """Render plain text / markdown-ish response."""
        if self.use_rich:
            try:
                self.console.print(Markdown(text))
            except Exception:
                self.console.print(text)
        else:
            print(text)

    def render_agent(self, agent: dict) -> None:
        """Render an agent profile card."""
        if self.use_rich:
            self._rich_agent(agent)
        else:
            self._plain_agent(agent)

    def render_task(self, task: dict) -> None:
        """Render a task detail card."""
        if self.use_rich:
            self._rich_task(task)
        else:
            self._plain_task(task)

    def render_inbox(self, tasks: list[dict], has_more: bool = False) -> None:
        """Render inbox task list."""
        if self.use_rich:
            self._rich_inbox(tasks, has_more)
        else:
            self._plain_inbox(tasks, has_more)

    def render_match_results(self, results: list[dict], summary: str = "") -> None:
        """Render match/search results."""
        if self.use_rich:
            self._rich_matches(results, summary)
        else:
            self._plain_matches(results, summary)

    def render_task_approval(self, task: dict) -> None:
        """Render a task approval card with action hints."""
        if self.use_rich:
            self._rich_approval(task)
        else:
            self._plain_approval(task)

    def render_status(self, agent: dict, health: Optional[dict] = None) -> None:
        """Render connection status."""
        if self.use_rich:
            self._rich_status(agent, health)
        else:
            self._plain_status(agent, health)

    def render_error(self, message: str) -> None:
        """Render an error message."""
        if self.use_rich:
            self.console.print(f"[bold red]Error:[/bold red] {message}")
        else:
            print(f"{_Colors.RED}{_Colors.BOLD}Error:{_Colors.RESET} {message}")

    def render_success(self, message: str) -> None:
        """Render a success message."""
        if self.use_rich:
            self.console.print(f"[bold green]OK:[/bold green] {message}")
        else:
            print(f"{_Colors.GREEN}{_Colors.BOLD}OK:{_Colors.RESET} {message}")

    def render_info(self, message: str) -> None:
        """Render an informational message."""
        if self.use_rich:
            self.console.print(f"[dim]{message}[/dim]")
        else:
            print(f"{_Colors.DIM}{message}{_Colors.RESET}")

    def render_tool_call(self, name: str, args: dict) -> None:
        """Show a tool call being made (debug mode)."""
        if self.use_rich:
            self.console.print(f"  [dim cyan]-> {name}({args})[/dim cyan]")
        else:
            print(f"  {_Colors.CYAN}{_Colors.DIM}-> {name}({args}){_Colors.RESET}")

    # ── Rich rendering ──

    def _rich_agent(self, agent: dict) -> None:
        table = Table(show_header=False, box=None, padding=(0, 2))
        table.add_column("Key", style="bold cyan", width=18)
        table.add_column("Value")

        table.add_row("Name", agent.get("display_name", "Unknown"))
        table.add_row("Slug", f"@{agent.get('slug', '?')}")
        table.add_row("Type", agent.get("agent_type", "?"))
        table.add_row("Status", self._status_badge(agent.get("status", "offline")))
        table.add_row("Verification", agent.get("verification_level", "none"))

        profile = agent.get("profile") or {}
        if profile.get("skills"):
            table.add_row("Skills", ", ".join(profile["skills"]))
        if profile.get("languages"):
            table.add_row("Languages", ", ".join(profile["languages"]))
        if profile.get("bio"):
            table.add_row("Bio", profile["bio"])
        if profile.get("location_city"):
            loc = profile.get("location_city", "")
            if profile.get("location_country"):
                loc += f", {profile['location_country']}"
            table.add_row("Location", loc)

        self.console.print(Panel(table, title="Agent Profile", border_style="blue"))

    def _rich_task(self, task: dict) -> None:
        table = Table(show_header=False, box=None, padding=(0, 2))
        table.add_column("Key", style="bold cyan", width=18)
        table.add_column("Value")

        table.add_row("Task ID", task.get("id", "?"))
        table.add_row("Type", task.get("task_type", "?"))
        table.add_row("Status", self._status_badge(task.get("status", "?")))
        table.add_row("Risk Level", self._risk_badge(task.get("risk_level", "R0")))
        if task.get("description"):
            table.add_row("Description", task["description"])
        table.add_row("From", task.get("from_agent_id", "?"))
        table.add_row("To", task.get("to_agent_id", "?"))
        if task.get("created_at"):
            table.add_row("Created", task["created_at"])

        self.console.print(Panel(table, title="Task Details", border_style="yellow"))

    def _rich_inbox(self, tasks: list[dict], has_more: bool) -> None:
        if not tasks:
            self.console.print("[dim]Inbox is empty.[/dim]")
            return

        table = Table(title="Inbox", show_lines=False)
        table.add_column("ID", style="dim", max_width=12)
        table.add_column("Status", max_width=20)
        table.add_column("Type", max_width=15)
        table.add_column("Risk")
        table.add_column("Description", max_width=40)

        for t in tasks:
            table.add_row(
                t.get("id", "?")[:12],
                self._status_badge(t.get("status", "?")),
                t.get("task_type", "?"),
                self._risk_badge(t.get("risk_level", "R0")),
                (t.get("description") or "")[:40],
            )

        self.console.print(table)
        if has_more:
            self.console.print("[dim]... more tasks available[/dim]")

    def _rich_matches(self, results: list[dict], summary: str) -> None:
        if summary:
            self.console.print(f"\n{summary}\n")

        if not results:
            self.console.print("[dim]No matches found.[/dim]")
            return

        for i, r in enumerate(results, 1):
            table = Table(show_header=False, box=None, padding=(0, 1))
            table.add_column("Key", style="bold", width=16)
            table.add_column("Value")

            table.add_row("Name", r.get("display_name", "Unknown"))
            table.add_row("Agent Ref", r.get("agent_ref", r.get("agent_id", "?")))
            if r.get("skills"):
                table.add_row("Skills", ", ".join(r["skills"]))
            if r.get("location"):
                table.add_row("Location", r["location"])
            if r.get("verification"):
                table.add_row("Verified", r["verification"])
            if r.get("last_active"):
                table.add_row("Last Active", r["last_active"])
            if r.get("success_rate_30d") is not None:
                table.add_row("Success Rate", f"{r['success_rate_30d']:.0%}")
            if r.get("why_matched"):
                for reason in r["why_matched"]:
                    table.add_row("", f"  - {reason}")
            if r.get("match_score") is not None:
                table.add_row("Score", f"{r['match_score']:.1f}")
            if r.get("reasons"):
                for reason in r["reasons"]:
                    table.add_row("", f"  - {reason}")

            self.console.print(Panel(table, title=f"Match #{i}", border_style="green"))

    def _rich_approval(self, task: dict) -> None:
        table = Table(show_header=False, box=None, padding=(0, 2))
        table.add_column("Key", style="bold cyan", width=18)
        table.add_column("Value")

        table.add_row("Task ID", task.get("id", "?"))
        table.add_row("Type", task.get("task_type", "?"))
        table.add_row("Risk Level", self._risk_badge(task.get("risk_level", "R0")))
        if task.get("description"):
            table.add_row("Description", task["description"])
        table.add_row("From", task.get("from_agent_id", "?"))

        panel_content = table
        self.console.print(Panel(panel_content, title="Task Approval Required", border_style="red"))
        self.console.print("  [bold green]/accept[/bold green] <id>  — Accept this task")
        self.console.print("  [bold red]/decline[/bold red] <id>  — Decline this task")

    def _rich_status(self, agent: dict, health: Optional[dict]) -> None:
        table = Table(show_header=False, box=None, padding=(0, 2))
        table.add_column("Key", style="bold cyan", width=18)
        table.add_column("Value")

        table.add_row("Agent", agent.get("display_name", "Unknown"))
        table.add_row("Slug", f"@{agent.get('slug', '?')}")
        table.add_row("Type", agent.get("agent_type", "?"))
        table.add_row("Status", self._status_badge(agent.get("status", "offline")))
        table.add_row("Verification", agent.get("verification_level", "none"))

        if health:
            table.add_row("Server", f"{health.get('service', '?')} v{health.get('version', '?')}")
            table.add_row("Region", health.get("region", "?"))

        self.console.print(Panel(table, title="Seabay Status", border_style="blue"))

    def _status_badge(self, status: str) -> str:
        color_map = {
            "online": "green",
            "offline": "dim",
            "busy": "yellow",
            "pending_accept": "yellow",
            "pending_delivery": "yellow",
            "delivered": "yellow",
            "accepted": "blue",
            "in_progress": "blue",
            "waiting_human_confirm": "yellow",
            "completed": "green",
            "failed": "red",
            "declined": "red",
            "cancelled": "dim",
            "expired": "dim",
        }
        color = color_map.get(status, "white")
        return f"[{color}]{status}[/{color}]"

    def _risk_badge(self, level: str) -> str:
        color_map = {"R0": "green", "R1": "yellow", "R2": "red", "R3": "bold red"}
        color = color_map.get(level, "white")
        return f"[{color}]{level}[/{color}]"

    # ── Plain text fallback rendering ──

    def _plain_agent(self, agent: dict) -> None:
        print(f"\n{'=' * 50}")
        print("  Agent Profile")
        print(f"{'=' * 50}")
        print(f"  Name:         {agent.get('display_name', 'Unknown')}")
        print(f"  Slug:         @{agent.get('slug', '?')}")
        print(f"  Type:         {agent.get('agent_type', '?')}")
        print(f"  Status:       {agent.get('status', 'offline')}")
        print(f"  Verification: {agent.get('verification_level', 'none')}")
        profile = agent.get("profile") or {}
        if profile.get("skills"):
            print(f"  Skills:       {', '.join(profile['skills'])}")
        if profile.get("languages"):
            print(f"  Languages:    {', '.join(profile['languages'])}")
        if profile.get("bio"):
            print(f"  Bio:          {profile['bio']}")
        print(f"{'=' * 50}\n")

    def _plain_task(self, task: dict) -> None:
        print(f"\n{'=' * 50}")
        print("  Task Details")
        print(f"{'=' * 50}")
        print(f"  ID:          {task.get('id', '?')}")
        print(f"  Type:        {task.get('task_type', '?')}")
        print(f"  Status:      {task.get('status', '?')}")
        print(f"  Risk:        {task.get('risk_level', 'R0')}")
        if task.get("description"):
            print(f"  Description: {task['description']}")
        print(f"  From:        {task.get('from_agent_id', '?')}")
        print(f"  To:          {task.get('to_agent_id', '?')}")
        print(f"{'=' * 50}\n")

    def _plain_inbox(self, tasks: list[dict], has_more: bool) -> None:
        if not tasks:
            print("\nInbox is empty.\n")
            return
        print(f"\nInbox ({len(tasks)} tasks):\n")
        for t in tasks:
            risk = f" [{t.get('risk_level', 'R0')}]" if t.get("risk_level", "R0") != "R0" else ""
            print(f"  {t.get('id', '?')[:12]}  {t.get('status', '?'):20s}  {t.get('task_type', '?'):15s}{risk}")
            if t.get("description"):
                print(f"    {t['description'][:80]}")
        if has_more:
            print("\n  ... more tasks available")
        print()

    def _plain_matches(self, results: list[dict], summary: str) -> None:
        if summary:
            print(f"\n{summary}\n")
        if not results:
            print("No matches found.\n")
            return
        for i, r in enumerate(results, 1):
            print(f"\n--- Match #{i} ---")
            print(f"  Name:     {r.get('display_name', 'Unknown')}")
            print(f"  Ref:      {r.get('agent_ref', r.get('agent_id', '?'))}")
            if r.get("skills"):
                print(f"  Skills:   {', '.join(r['skills'])}")
            if r.get("why_matched"):
                for reason in r["why_matched"]:
                    print(f"    - {reason}")
            if r.get("reasons"):
                for reason in r["reasons"]:
                    print(f"    - {reason}")
        print()

    def _plain_approval(self, task: dict) -> None:
        print(f"\n{'=' * 50}")
        print("  TASK APPROVAL REQUIRED")
        print(f"{'=' * 50}")
        print(f"  Task ID:     {task.get('id', '?')}")
        print(f"  Type:        {task.get('task_type', '?')}")
        print(f"  Risk:        {task.get('risk_level', 'R0')}")
        if task.get("description"):
            print(f"  Description: {task['description']}")
        print(f"\n  /accept {task.get('id', '<id>')}  — Accept this task")
        print(f"  /decline {task.get('id', '<id>')} — Decline this task")
        print(f"{'=' * 50}\n")

    def _plain_status(self, agent: dict, health: Optional[dict]) -> None:
        print(f"\n{'=' * 50}")
        print("  Seabay Status")
        print(f"{'=' * 50}")
        print(f"  Agent:        {agent.get('display_name', 'Unknown')}")
        print(f"  Slug:         @{agent.get('slug', '?')}")
        print(f"  Status:       {agent.get('status', 'offline')}")
        if health:
            print(f"  Server:       {health.get('service', '?')} v{health.get('version', '?')}")
        print(f"{'=' * 50}\n")
